design/sketch.webidl

```diff
@@ -105,16 +105,51 @@ interface WebGPUTexture {
     WebGPUTextureView createTextureView(WebGPUTextureViewDescriptor desc);
 };
 
-// Sampler
+// Samplers
+enum WebGPUAddressMode {
+    "clampToEdge",
+    "mirrorClampToEdge",
+    "repeat",
+    "mirrorRepeat",
+    "clampToBorderColor"
+}
+
 enum WebGPUFilterMode {
     "nearest",
-    "linear",
-};
+    "linear"
+}
+
+enum WebGPUCompareFunction {
+    "never",
+    "less",
+    "equal",
+    "lessEqual",
+    "greater",
+    "notEqual",
+    "greaterEqual",
+    "always"
+}
+
+enum WebGPUBorderColor {
+    "transparentBlack",
+    "opaqueBlack",
+    "opaqueWhite"
+}
 
 dictionary WebGPUSamplerDescriptor {
-    WebGPUFilterModeEnum magFilter;
-    WebGPUFilterModeEnum minFilter;
-    WebGPUFilterModeEnum mipmapFilter;
+    WebGPUddressMode rAddressMode = "clampToEdge",
+    WebGPUddressMode sAddressMode = "clampToEdge",
+    WebGPUddressMode tAddressMode = "clampToEdge",
+    WebGPUFilterModeEnum magFilter = "nearest";
+    WebGPUFilterModeEnum minFilter = "nearest";
+    WebGPUFilterModeEnum mipmapFilter = "nearest";
+    float lodMinClamp = 0,
+    float lodMaxClamp = FLT_MAX,
+    unsigned long maxAnisotropy = 1,
+    boolean normalizedCoordinates = true,
```

Comment:
I didn't see something like this in D3D12_SAMPLER_DESC.

Reply:
Yep, you’re right, I missed that. It would have to be emulated, too.

Reply:
Perhaps, we should instead remove this for now?